### PR TITLE
Added getInitialJukeboxState to app.js

### PR DESCRIPTION
**Pull request by peddybeats**

### CODE REVIEW

File: app/src/app.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+module.exports = {
+  getInitialJukeboxState: getInitialJukeboxState
+};
+
+/*
+ * Example usage of getInitialJukeboxState()
+ */
+function main() {
+  accessToken =  
+    'BQD4gqZ79fYkOFjIQwiBPaUtKJLFlaN58lspY5zmtvY9T8RSqMWVq-LMSvTocb4Lvn-cuOkzOyDI9E8zi5s3MR1k90aP4yTfN9UuZTEu6ttWCorX5GM4f7csSNe86ZM6WKwKuPg8C9sS50gGUmsDGQ';
```
Comment (peddybeats) [inline app/src/app.js]: Take out

File: app/src/app.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  var numTracks = tracksFeatures.body.audio_features.length;
+
+  // Aggregate every feature for every track
+  tracksFeatures.body.audio_features.forEach(function(trackFeatures) {
+    Object.keys(features).forEach(function(feature) {
+      features[feature] += trackFeatures[feature];
+    });   
+  });
+
+  // Take simply average of all features
```
Comment (peddybeats) [inline app/src/app.js]: s/simply/a\ simple

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
+*.swp
```
Comment (peddybeats) [inline .gitignore:3]: Add temp file formats for other editors as well

File: app/src/app.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
+var SpotifyWebApi = require('spotify-web-api-node')
```
Comment (peddybeats) [inline app/src/app.js]: nit: semicolon

File: app/src/app.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+function getTopTracks(accessToken) {
+  // Instatiate api instance and set its accessToken
+  var spotifyApi = new SpotifyWebApi();
+  spotifyApi.setAccessToken(accessToken);
+
+  // Return top tracks wrapped in a promise
+  return spotifyApi.getMyTopTracks();
+}
+
+function getAudioFeatures(accessToken, trackIds) {
```
Comment (peddybeats) [inline app/src/app.js]: Add doc

File: app/src/app.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+ * Returns a promise containing default Jukebox parameters
+ */
+function getDefaultJukeboxParams(tracksFeatures) {
+  features = {
+    'acousticness' : 0,
+    'danceability' : 0,
+    'energy' : 0,
+    'tempo' : 0,
+    'valence' : 0
+  }
```
Comment (peddybeats) [inline app/src/app.js]: nit: semicolon

File: app/src/app.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+ * Returns a promise containing select fields of potential seed tracks
+ */
+function getSeedTracks(topTracksObj) {
+  seedTracks = [];
+  fields = {
+    'album': null,
+    'artists': [],
+    'name': null,
+    'id': null,
+  };
```
Comment (peddybeats) [inline app/src/app.js]: Extract into constants.js

File: app/src/app.js (hunk 6 above)
Comment (peddybeats) [inline app/src/app.js]: Extract into constants.js

File: .gitignore (hunk 3 above)
Comment (peddybeats) [reply .gitignore:3]: Done

File: app/src/app.js (hunk 4 above)
Comment (peddybeats) [reply app/src/app.js]: Done

File: app/src/app.js (hunk 5 above)
Comment (peddybeats) [reply app/src/app.js]: Done

File: app/src/app.js (hunk 6 above)
Comment (peddybeats) [reply app/src/app.js]: Done

**Review by mira-holford — APPROVED** [review]

File: app/src/app.js (hunk 6 above)
Comment (peddybeats) [reply app/src/app.js]: Done but included it in app.js

File: app/src/app.js (hunk 2 above)
Comment (peddybeats) [reply app/src/app.js]: Done

File: app/src/app.js (hunk 7 above)
Comment (peddybeats) [reply app/src/app.js]: Done but included it in app.js

File: app/src/app.js (hunk 1 above)
Comment (peddybeats) [reply app/src/app.js]: Done

**Review by peddybeats — COMMENTED** [review]
LGTM

**Pull request merged**